app/models/news_articles.rb

```diff
@@ -27,15 +27,29 @@ def get_methods
   end
 
   def next
-    data = ExternalDatum.first.data
+    if ExternalDatum.source_types.include? metadata['source']
+      data = ExternalDatum.find_by(source_type: metadata['source']).data
+    else
+      data = ExternalDatum.first.data
+    end
     index = (self.metadata.dig('id').to_i + 1 < data.length) ? metadata.dig('id').to_i + 1 : 0
     update_board data.dig(index, 'title'), data.dig(index, 'href'), index
     broadcast
     sync_data
   end
 
   def update_board value, href, id
-    update! metadata: { value: value, id: id, href: href }
+    m = metadata
+    m['value'] = value
+    m['id'] = id
+    m['href'] = href
+    update! metadata: m
```

Comment:
pai pare ca e dar, defapt  nu e :)). Ce se intample e ca daca zic update metadata: { value => value }, dupa asta metadata va contine doar fieldul cu value. pe cand daca fac ca mai sus, toate celelalte fielduri care existau inainte de update, raman asa cum erau. daca nu se sterge tot si ramane doar value: value in metadata